src/clue.js

```diff
@@ -2,25 +2,204 @@
+//         firstName: "Eleanor",
+//         lastName: "Peacock",
+//         occupation: "Socialité",
+//         age: 36,
+//         description: "She is from a wealthy family and uses her status and money to earn popularity",
+//         image: "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+//         color: "blue"},
+
+//     let mrMustard= {
+//         firstName: "Jack",
+//         lastName: "Mustard",
+//         occupation: "Retired Football player",
+//         age: 62,
+//         description: "He is a former football player who tries to get by on his former glory",
+//         image: "https://static.independent.co.uk/s3fs-public/thumbnails/image/2016/07/04/08/unspecified-3.jpg",
+//         color: "yellow"}
+   
 
 // Rooms Array
 
-const roomsArray = [];
+const roomsArray = [
+    {name: "Dining Room"},
+        {name: "Conservatory"},
+        {name: "Kitchen"},
+        {name: "Study"},
+        {name: "Library"},
+        {name: "Billiard Room"},
+        {name: "Lounge"},
+        {name: "Ballroom"},
+        {name: "Hall"},
+        {name: "Spa"},
+        {name: "Living Room"},
+        {name: "Observatory"},
+        {name: "Theater"},
+        {name: "Guest House"},
+        {name: "Patio"}
+];
 
 // Weapons Array
 
-const weaponsArray = [];
+const weaponsArray = [
+    {name: "rope", weight: 10},
+        {name: "knife", weight: 8},
+        {name: "candlestick", weight: 2},
+        {name: "dumbbell", weight: 30},
+        {name: "poison", weight: 2},
+        {name: "axe", weight: 15},
+        {name: "bat", weight: 13},
+        {name: "trophy", weight: 25},
+        {name: "pistol", weight: 20}
+];
 
 
 // ITERATION 2
 
-function selectRandom() {}
+// const result = selectRandom(roomsArray)
+// console.log(result)
+
+function selectRandom(arr) {
+    const randomIndex = Math.floor(Math.random() * arr.length);
+    const item = arr[randomIndex];
+    return item;
+}
+
+function pickMystery(arr) {
 
-function pickMystery() {}
+    const mystery = {
+        suspect: selectRandom(suspectsArray),
+        weapon: selectRandom(weaponsArray),
+        room: selectRandom(roomsArray)
+    }
+  return mystery
+}
 
+  // const randomIndex = Math.floor(Math.random() * roomArr.length);
+    // const randomCard = roomArr[randomIndex];
+    // return randomCard;
+    // let room = pickMystery(roomsArray)
+    // console.log(room)
+
+
+    // let weapon = pickMystery(weaponsArray)
+    // console.log(weapon)
+
+
+    // let suspect = pickMystery(suspectsArray)
+    // console.log(suspect)
+
+    
+    
+
+
+    // let endResult = {}
+    // endResult.room = "Library"
+    // endResult.weapon = "Candlestick"
+    // endResult.suspect = "mrMustard"
+    // console.log(room.push)
 
 // ITERATION 3
 
-function revealMystery() {}
+function revealMystery (object){
+    return `${object.suspect.firstName} ${object.suspect.lastName} killed Mr. Boddy using the ${object.weapon.name} in the ${object.room.name}!`
```

Comment:
Here, you have everything correct, however, it is good practice if you destructure your `object` argument.
By destructuring, it clearly expresses which properties you're using within the function.
By extracting the properties into individual variables (suspect, weapon, and room), you can directly reference them in your return statement without needing to use `object.suspect`, `object.weapon`, and `object.room` every time. This simplifies the code and improves readability.
It would look something like this:
```
function revealMystery(object) {
  let { suspect, weapon, room } = object;

  return `${suspect.firstName} ${suspect.lastName} killed Mr. Boddy using the ${weapon.name} in the ${room.name}!`;
}
```